### DISCUSSION TITLE
Testing gotcha

### DISCUSSION THREAD

**Discussion by hms** [shrinerb/shrine#690] [General]
When testing with the memory storage, it's easy to forget that without specifically deleting Shrine resources between tests, the store will likely retain every cached and stored file for the duration of the entire test suite, which can grow to a significant memory footprint.

Obviously, one can handle this manually per test or try to remember which classes of tests use Shrine. 

But if you prefer to just make sure it "works", this will do the trick for RSpec users (sorry, don't know the equivalent mini-test functionality):

```ruby
RSpec.configure do |config|
  config.after { Shrine.storages.each_value(&:clear!) }
end
```

Since the memory store is just as hash, the performance penalty for doing this indiscriminately per test rather than only for the tests that use Shrine is hardly noticeable.  But if that is a concern, the line above can be optimized with:

```ruby
RSpec.configure do |config|
  config.after(:example, shrine: true) { Shrine.storages.each_value(&:clear!) }
end
```

and then one just has to annotate tests / test classes that use Shrine with:

```ruby
RSpec.describe Class, shrine: true do ... end
```

**Comment by jrochkind** [comment]
I thought this seemed like an important tip, thanks for mentioning!  I went to look at my spec_helper... and saw I already had it there!  Git blame says 5 years ago. I wonder where I got the advice initially, perhaps it's already documented somewhere?

**Reply by hms** [reply]
I did look for this issue and solution before posting.  But, this could be another case of what my wife calls "daddy vision" where what is in front of me is invisible.

**Reply by jrochkind** [reply]
It's def possible it's not there and I just figured it out on my own, or from another discussion!

If you find a place in docs you think it belongs, if you PR it maybe janko will merge?
